File: linmour-order-module/linmour-order-api/src/main/java/com/linmour/order/pojo/Do/OrderInfo.java
package com.linmour.order.pojo.Do;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;


/**
 * @TableName order_info
 */
@TableName(value = "order_info")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderInfo implements Serializable {
    /**
     *
     */
    @TableId(value = "id")
    private String id;

    /**
     * 用户id
     */
    @TableField(value = "openid")
//    @Column("openid")
    private String openid;

    @TableField(value = "table_id")
//    @Column("table_id")
    private Long tableId;

    /**
     * 支付方式 1支付宝 2微信 3积分..
     */
    @TableField(value = "pay_type")
//    @Column(value = "pay_type")
    private Integer payType;

    /**
     * 支付状态 1已支付 2未支付 3已取消
     */
    @TableField(value = "pay_status")
//    @Column(value = "pay_status")
    private Integer payStatus;

    /**
     * 支付金额
     */
    @TableField(value = "pay_amount")
//    @Column(value = "pay_amount")
    private BigDecimal payAmount;

    /**
     * 订单状态 1已完成 2进行中 3待处理
     */
    @TableField(value = "order_status")
//    @Column(value = "order_status")
    private Integer orderStatus;

    /**
     * 备注
     */
    @TableField(value = "remark")
//    @Column(value = "remark")
    private String remark;

    @TableField(value = "shop_id")
//    @Column(value = "shop_id")
    private Long shopId;


    /**
     *
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(value = "create_time", fill = FieldFill.INSERT)
//    @Column(value = "create_time")
    private LocalDateTime createTime;

    /**
     *
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
//    @Column(value = "update_time")
    private LocalDateTime updateTime;

    /**
     *
     */
    @TableField(value = "create_by", fill = FieldFill.INSERT)
//    @Column(value = "create_by")
    private Long createBy;

    /**
     *
     */
    @TableField(value = "update_by", fill = FieldFill.INSERT_UPDATE)
//    @Column(value = "update_by")
    private Long updateBy;

    /**
     *
     */
    @TableField(value = "deleted")
//    @Column(value = "deleted")
    private Integer deleted;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}

File: linmour-order-module-go/server/models/OrderInfo.go
package models

import (
	"encoding/json"
	"strconv"
)

// OrderInfo 代表订单信息的结构体
type OrderInfo struct {
	ID          string  `json:"id" gorm:"column:id"`                    // 订单ID
	Openid      string  `json:"openid" gorm:"column:openid"`            // 用户ID
	TableId     int64   `json:"tableId" gorm:"column:table_id"`         // 桌号ID
	PayType     int32   `json:"payType" gorm:"column:pay_type"`         // 支付方式
	PayStatus   int32   `json:"payStatus" gorm:"column:pay_status"`     // 支付状态
	PayAmount   float64 `json:"payAmount" gorm:"column:pay_amount"`     // 支付金额
	OrderStatus int32   `json:"orderStatus" gorm:"column:order_status"` // 订单状态
	Remark      string  `json:"remark" gorm:"column:remark"`            // 备注
	ShopId      int64   `json:"shopId" gorm:"column:shop_id"`           // 商店ID
	CreateTime  string  `json:"createTime" gorm:"column:create_time"`   // 创建时间
	UpdateTime  string  `json:"updateTime" gorm:"column:update_time"`   // 更新时间
	CreateBy    string  `json:"createBy" gorm:"column:create_by"`       // 创建人
	UpdateBy    string  `json:"updateBy" gorm:"column:update_by"`       // 更新人
	Deleted     string  `json:"deleted" gorm:"column:deleted"`          // 是否删除
}

// UnmarshalJSON 自定义JSON解码逻辑
func (o *OrderInfo) UnmarshalJSON(data []byte) error {
	// 创建一个中间结构体来接收JSON数据
	type intermediate struct {
		ID          string `json:"id"`
		Openid      string `json:"openid"`
		TableId     string `json:"tableId"`
		PayType     string `json:"payType"`
		PayStatus   string `json:"payStatus"`
		PayAmount   string `json:"payAmount"`
		OrderStatus string `json:"orderStatus"`
		Remark      string `json:"remark"`
		ShopId      string `json:"shopId"`
		CreateTime  string `json:"createTime"`
		UpdateTime  string `json:"updateTime"`
		CreateBy    string `json:"createBy"`
		UpdateBy    string `json:"updateBy"`
		Deleted     string `json:"deleted"`
	}

	var jsonMap intermediate
	err := json.Unmarshal(data, &jsonMap)
	if err != nil {
		return err
	}

	// 手动转换字段类型
	o.ID = jsonMap.ID
	o.Openid = jsonMap.Openid
	o.TableId, err = strconv.ParseInt(jsonMap.TableId, 10, 64)
	if err != nil {
		return err
	}
	payTypeInt, err := strconv.ParseInt(jsonMap.PayType, 10, 32)
	if err != nil {
	}
	// 显式转换为 int32 并赋值给 o.PayType
	o.PayType = int32(payTypeInt)

	// 解析 PayStatus 字段
	payStatusInt, err := strconv.ParseInt(jsonMap.PayStatus, 10, 32)
	if err != nil {
	}
	// 显式转换为 int32 并赋值给 o.PayStatus
	o.PayStatus = int32(payStatusInt)
	o.PayAmount, err = strconv.ParseFloat(jsonMap.PayAmount, 64)
	if err != nil {
		return err
	}
	orderStatus, err := strconv.ParseInt(jsonMap.OrderStatus, 10, 32)
	if err != nil {
		// 设置默认的 OrderStatus，例如 0
		o.OrderStatus = int32(0)
	} else {
		o.OrderStatus = int32(orderStatus)
	}
	o.Remark = jsonMap.Remark
	o.ShopId, err = strconv.ParseInt(jsonMap.ShopId, 10, 64)
	if err != nil {
		return err
	}
	o.CreateTime = jsonMap.CreateTime
	o.UpdateTime = jsonMap.UpdateTime

	o.CreateBy = jsonMap.CreateBy
	o.UpdateBy = jsonMap.UpdateBy
	o.Deleted = jsonMap.Deleted

	return nil
}
